parquet-cli/src/main/java/org/apache/parquet/cli/commands/ScanCommand.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.cli.commands;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.io.Closeables;
+import org.apache.avro.Schema;
+import org.apache.parquet.cli.BaseCommand;
+import org.apache.parquet.cli.util.Expressions;
+import org.slf4j.Logger;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.List;
+
+@Parameters(commandDescription = "Scan all records from a file")
+public class ScanCommand extends BaseCommand {
+
+  @Parameter(description = "<file>")
+  List<String> sourceFiles;
+
+  @Parameter(
+    names = {"-c", "--column", "--columns"},
+    description = "List of columns")
+  List<String> columns;
+
+  public ScanCommand(Logger console) {
+    super(console);
+  }
+
+  @Override
+  public int run() throws IOException {
+    Preconditions.checkArgument(
+      sourceFiles != null && !sourceFiles.isEmpty(),
+      "Missing file name");
+    Preconditions.checkArgument(sourceFiles.size() == 1,
+      "Only one file can be given");
+
+    final String source = sourceFiles.get(0);
+    Schema schema = getAvroSchema(source);
+    Schema projection = Expressions.filterSchema(schema, columns);
+
+    long startTime = System.currentTimeMillis();
+    Iterable<Object> reader = openDataFile(source, projection);
+    boolean threw = true;
+    long count = 0;
+    try {
+      for (Object record : reader) {
+        count += 1;
```

Comment:
If your goal is only to get count, why not get it from the metadata? Iterate each record is an expensive operation.

Reply:
It serves the same purpose as https://github.com/apache/arrow/blob/master/cpp/tools/parquet/parquet_scan.cc to validate the data integrity of a parquet file.

Reply:
This is very handy when we want to quickly check if any file is corrupted in production.